frontend/widgets/OBSBasic_Transitions.cpp

```diff
@@ -399,13 +402,10 @@ void OBSBasic::SetTransition(OBSSource transition)
 
 OBSSource OBSBasic::GetCurrentTransition()
 {
-	return ui->transitions->currentData().value<OBSSource>();
-}
+	if (transitions.find(currentTransitionUuid) == transitions.end())
+		return nullptr;
 
-void OBSBasic::on_transitions_currentIndexChanged(int)
-{
-	OBSSource transition = GetCurrentTransition();
-	SetTransition(transition);
+	return transitions[currentTransitionUuid];
```

Comment:
```suggestion

auto currentTransition = transitions.find(currentTransitionUuid);

if (currentTransition == transitions.end()) {
    return nullptr;
}

return currentTransition->second;
```

Iterator already yields direct access to the element in the map, using subscript operator just iterates over the map again and potentially _inserts_ a non-existent element.

Reply:
Mutators removed